tests/integration/helpers.py

```diff
@@ -159,18 +159,28 @@ async def mongodb_uri(
     use_subprocess_to_get_password=False,
     port=MONGOD_PORT,
     app_name: str = APP_NAME,
+    username: str = "operator",
+    password: str | None = None,
 ) -> str:
     if unit_ids is None:
         unit_ids = range(0, len(ops_test.model.applications[app_name].units))
 
     addresses = [await get_address_of_unit(ops_test, unit_id, app_name) for unit_id in unit_ids]
     hosts = [f"{host}:{port}" for host in addresses]
     hosts = ",".join(hosts)
-    if use_subprocess_to_get_password:
-        password = get_password_using_subprocess(ops_test, app_name=app_name)
-    else:
-        password = await get_password(ops_test, 0, app_name=app_name)
-    return f"mongodb://operator:{password}@{hosts}/admin"
+
+    # If password is chosen, always keep it, otherwise check on `use_subprocess_to_get_password`
+    match (password, use_subprocess_to_get_password):
+        case (None, True):
+            password = get_password_using_subprocess(
+                ops_test, username=username, app_name=app_name
+            )
+        case (None, False):
+            password = await get_password(ops_test, 0, username=username, app_name=app_name)
+        case _:
+            pass
```

Comment:
If there is no username + password the URI should look like `f"mongodb://{hosts}/admin"`

might need to update this to 
```suggestion
            return `f"mongodb://{hosts}/admin"`
```

Reply:
That's not exactly it but I agree I can already return the right URI here, but I won't do it for the sake of readability.
My checks are the following:
* If I have no password and I specify I should go through a subprocess to get it, I do that
* If I have no password and I specify I should run the action to get the password, I do that
* Otherwise, whatever happens I provided the password and I'll use that directlY